msfvenom

```diff
@@ -69,81 +69,69 @@ def parse_args(args)
   opt.separator('')
   opt.separator('Options:')
 
+  opt.on('-l', '--list            <type>', Array, 'List all module for [type]. Types are: payloads, encoders, nops, platforms, encrypt, formats, all') do |l|
+    if l.nil? or l.empty?
+      l = ["all"]
+    end
+    opts[:list] = l
+  end
+
   opt.on('-p', '--payload         <payload>', String,
-         'Payload to use. Specify a \'-\' or stdin to use custom payloads') do |p|
+         'Payload to use (--list payloads to list, --list-options for arguments). Specify \'-\' or STDIN for custom') do |p|
     if p == '-'
       opts[:payload] = 'stdin'
     else
       opts[:payload] = p
     end
   end
 
-  opt.on('--payload-options', "List the payload's standard options") do
+  opt.on('--list-options', "List --payload <value>'s standard, advanced and encryption options") do
     opts[:list_options] = true
   end
 
-  opt.on('-l', '--list            [type]', Array, 'List a module type. Options are: payloads, encoders, nops, all') do |l|
-    if l.nil? or l.empty?
-      l = ["all"]
-    end
-    opts[:list] = l
-  end
-
-  opt.on('-n', '--nopsled         <length>', Integer, 'Prepend a nopsled of [length] size on to the payload') do |n|
-    opts[:nops] = n.to_i
-  end
-
-  opt.on('-f', '--format          <format>', String, "Output format (use --help-formats for a list)") do |f|
+  opt.on('-f', '--format          <format>', String, "Output format (use --list formats to list)") do |f|
     opts[:format] = f
   end
 
-  opt.on('--help-formats', String, "List available formats") do
-    init_framework(:module_types => [])
-    msg = "Executable formats\n" +
-      "\t" + ::Msf::Util::EXE.to_executable_fmt_formats.join(", ") + "\n" +
-      "Transform formats\n" +
-      "\t" + ::Msf::Simple::Buffer.transform_formats.join(", ")
-    raise HelpError, msg
+  opt.on('-e', '--encoder         <encoder>', String, 'The encoder to use (use --list encoders to list)') do |e|
+    opts[:encoder] = e
   end
 
-  opt.on('--encrypt         <value>', String, 'The type of encryption or encoding to apply to the shellcode') do |e|
-    opts[:encryption_format] = e
+  opt.on('--smallest', 'Generate the smallest possible payload using all available encoders') do
+    opts[:smallest] = true
   end
 
-  opt.on('--encrypt-formats', String, 'List Available encryption formats') do
-    init_framework(:module_types => [])
-    msg = "Encryption formats:\n" +
-          "\t" + ::Msf::Simple::Buffer.encryption_formats.join(", ")
-    raise HelpError, msg
+  opt.on('--encrypt         <value>', String, 'The type of encryption or encoding to apply to the shellcode (use --list encrypt to list)') do |e|
+    opts[:encryption_format] = e
   end
 
-  opt.on('--encrypt-key     <value>', String, 'A key to be used for the encryptor') do |e|
+  opt.on('--encrypt-key     <value>', String, 'A key to be used for --encrypt') do |e|
```

Comment:
I would like to see which `--encrypt-key` and `encrypt-iv` options are required for each encryption type be referenced as required by the `--list encrypt` output.

Not needed to land this but something I think would be helpful to do.

Reply:
Thanks for the suggestion.
Added it to my wish list ideas ;)